Add `namespaceSelector` inputs to Cluster Admission Policies

jordojordo opened the pull request:
## Description

<!-- Please provide the link to the GitHub issue you are addressing -->
Fix #394

<!-- Please provide the link to the documentation related to your change, if applicable -->
<!-- [Documentation](https://<insert your url>) -->
This adds inputs for `matchExpressions` and `matchLabels` for the ClusterAdmissionPolicy spec when creating or editing a policy. They will be under the "General" tab of a policy's config.

Also adds unit tests for the inputs to ensure labels and expressions are added/removed to the policy's spec.

https://github.com/kubewarden/kubewarden-ui/assets/40806497/0da3058d-ccc0-4b9d-9b28-147b9a5eb349

viccuad commented on pkg/kubewarden/l10n/en-us.yaml:
```diff
@@ -176,6 +176,15 @@ kubewarden:
       tooltip: Certain policies will break core services of Rancher, this will add a default list of namespaces to ignore.
+    namespaceSelector:
+      label: Namespace Selector
+      tooltip: NamespaceSelector decides whether to run the webhook on an object based on whether the namespace for that object matches the selector. If the object itself is a namespace, the matching is performed on object.metadata.labels. If the object is another cluster scoped resource, it never skips the webhook.
```
```suggestion
      tooltip: NamespaceSelector decides whether to evaluate the policy on a request, based on whether the namespace for that request matches the selector. If the object of the request is a namespace, the matching is performed on object.metadata.labels. If the object is another cluster scoped resource, it never skips the policy evaluation.
```

Trying to improve on the docs that come from https://doc.crds.dev/github.com/kubewarden/kubewarden-controller/policies.kubewarden.io/ClusterAdmissionPolicy/v1@v1.7.0-rc1#spec-namespaceSelector.

(Those docs get their doc string from the upstream namespaceSelector, which we don't want to touch for compatibility reasons. But that means they are not explicative if you don't know what a namespaceSelector is beforehand)

viccuad reviewed (approved):
LGTM besides the doc string nit.

flavio commented:
Looks really good :clap: 

Would it be possible to start with no pre-populated selectors? From what I see from the video there are already some match expression set.

From a UX POV, I wonder how we can ensure the user is not scared about leaving the namespace selectors unconfigured. Do you think this is already clear? Should we use a toggle switch to enable namespace selector and then have the user enter his rules? :thinking: 

Maybe Ken can help with that

viccuad commented:
I agree with Flavio, it may scare the user. I prefer to have a toggle switch than to have it empty.

jordojordo commented:
The reason for those initial `matchExpressions` are due to the "Ignore Rancher Namespaces" radio button further up on the form. We are making this true by default and those are added automatically. 
We could make the "Ignore Rancher Namespaces" button false by default - If I remember correctly we went with true by default to alleviate any issues for users .
I'm not opposed to moving the entire Namespace Selector to be behind a button, or perhaps a tab on it's own.

viccuad commented:
I like the idea of them being in their own NamespaceSelector tab. I would put the "Ignore Rancher Namespaces" button there, true by default, and at least is less shocking for a first time user.

Also, I notice that for adding a new selector rule, it says "Add Rule", which, in the general tab, may be confused with the policy `spec.rules`. Moving it to its own tab helps with this. Maybe worth renaming that "Add Rule" to "Add matchExpression" or "Add expresion".

jordojordo commented:
@viccuad Good points, I'll move those things into a separate tab and update the wordings a bit.

jordojordo commented:
I've updated the Namespace Selector section to be within it's own tab as well as behind a radio button to remove any confusion for new users. Also updated the Namespace Selector description to match suggested changes.

![1](https://github.com/kubewarden/kubewarden-ui/assets/40806497/69fd14b7-9822-4b77-a84c-b3ae5f6e72c7)

![2](https://github.com/kubewarden/kubewarden-ui/assets/40806497/298ff0bc-1f2e-4d65-bc49-abd6947b663d)

flavio commented:
That looks better, just one note. Shouldn't we use a toggle button instead of the "yes/no" ones?

Something like that

![image](https://github.com/kubewarden/kubewarden-ui/assets/22728/0bb7cfb4-e02c-4653-a289-7cc15c65f238)

viccuad reviewed (approved):
LGTM. Agree with Flavio, if possible, I would prefer a toggle button.

jordojordo commented:
Rancher does not use those types of toggle buttons. However, I could turn into a check box that would be more akin to how we enable/disable global settings. For example, in the Branding section:
 

https://github.com/kubewarden/kubewarden-ui/assets/40806497/0b8bafe2-841f-43a0-be19-d3bd5a7fcdc8

flavio commented:
@jordojordo that looks better! :+1:

viccuad commented:
what about the "Ignore Rancher Namespaces" one? does it make sense for it be a checkbox?
I should shut up and stop yak shaving..

flavio commented:
I think this looks good, as @viccuad maybe we should use a checkbox for the "ignore rancher namespaces", but I don't know what the UX guidelines are for rancher

jordojordo commented:
I can certainly change it to a checkbox, it would probably make more sense to have some consistency within this tab. @kwwii What do you suggest here?

jordojordo commented:
I went ahead with updating the RadioButton into a Checkbox, I think this is a nicer approach for the UX.

![1](https://github.com/rancher/kubewarden-ui/assets/40806497/eab8e77b-2132-4aca-8954-86fcb18f21a1)
![2](https://github.com/rancher/kubewarden-ui/assets/40806497/a0be2395-36b1-46d5-bf66-6dd3d21e1832)

flavio commented:
that looks great to me, :ship: it!

flavio commented:
@Martin-Weiss JFYI: this implements a feature you requested. It's going to be part of the next release of the kubewarden-ui
